src/main/java/graphql/GraphQL.java

```diff
@@ -188,11 +191,58 @@ public ExecutionResult execute(String requestString, Object context, Map<String,
     }
 
     public ExecutionResult execute(String requestString, String operationName, Object context, Map<String, Object> arguments) {
```

Comment:
I left it in to allow easier migration BUT I think we should break it

that is implementors of ExecutionStrategy are required to produce promises

We can keep the outer ones in place as "sugar" for people who just want a ExecutionResult direct